### ISSUE TITLE
metatensor supports `astype(dtype, device)`

### ISSUE THREAD

**Issue by wyli** [Project-MONAI/MONAI#4636]
**Is your feature request related to a problem? Please describe.**
would be great to support `astype` API like numpy's astype https://numpy.org/doc/stable/reference/generated/numpy.ndarray.astype.html

`astype(dtype)` the dtype could be numpy based (and detach if needed when it's from a torch.tensor), as well as torch based.
